snap/snapcraft.yaml

```diff
@@ -0,0 +1,81 @@
+name: deliveryoptimization-client
+base: core20 # the base snap is the execution environment for this snap
+version: '0.1'
+summary: Ubuntu Core 20.04 DO
+description: |
+  A proof-of-concept for the Delivery Optimization Client Ubuntu Core snap.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: devmode # use 'strict' once you have the right plugs and slots
+
+#####
+#
+# Keywords
+#
+# after  - Specifies part's dependencies.
+#              See https://snapcraft.io/docs/parts-lifecycle#heading--step-dependencies
+#                  https://snapcraft.io/docs/parts-lifecycle
+# plugin - Specifies a plugin required for building the part.
+#
+####
+
+parts:
+  installdeps:
+    plugin: nil
+    source: .
+    override-build: |
+      ./build/scripts/bootstrap.sh --platform ubuntu2004 --install build
+
+  agent:
+    plugin: python
+    source: .
+    override-build: |
+      python3 ./build/build.py --project agent --ubuntu-core-snap-only
```

Comment:
This can come later but something to keep in mind before calling this complete: figure out how to build debug and minsizerel versions for snap.